nix/default.nix

```diff
@@ -0,0 +1,55 @@
+{ system ? builtins.currentSystem }:
+let
+  pkgs = (import ./nixpkgs.nix {
+    config = {
+      packageOverrides = pkg: {
+        systemd = pkg.systemd.overrideAttrs(x: {
+          postInstall = ''
+            ${x.postInstall}
+            ninja libsystemd.a
+            mv libsystemd.a $lib/lib/libsystemd.a
+          '';
+          postFixup = ''
+            ${x.postFixup}
+            sed -ri "s;$out/(.*);$nukedRef/\1;g" $lib/lib/libsystemd.a
+          '';
+        });
+      };
+    };
+  });
+
+  self = with pkgs; {
+    crun-static = (crun.overrideAttrs(x: {
+      name = "crun-static";
+      src = ./..;
+      nativeBuildInputs = [ autoreconfHook pkgconfig python3 ];
+      buildInputs = x.buildInputs ++ [ glibc glibc.static ];
+      preConfigure = ''
+        export LDFLAGS="-static-libgcc -static"
+        export CRUN_LDFLAGS="-all-static"
+        export LIBS="${yajl}/lib/libyajl_s.a"
```

Comment:
If I add a path to libsystemd.a here or not, I get this error in any case:
```
checking systemd/sd-bus.h usability... yes
checking systemd/sd-bus.h presence... yes
checking for systemd/sd-bus.h... yes
checking for library containing sd_listen_fds... no
configure: error: *** Failed to find libsystemd
```
@giuseppe is there a way how I can enforce the usage of the libsystemd.a file while still having the shared object in the same path?

Reply:
I think we need to add it to `LIBS=`.

We also need `librt.a` and `libpthread.a` as they are used by libsystemd.

I've tried to do it locally but I've not managed to get it working.

What is the correct way to add, let's say, `librt.a` from glibc.static?

Reply:
How did we make it works for CRI-O? I couldn’t find hits from:

- https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/virtualisation/cri-o.nix
- https://github.com/cri-o/cri-o/blob/master/nix/default.nix

Reply:
@saschagrunert could this be a hints?

https://github.com/containers/crun/blob/master/contrib/static-builder-x86_64/build.sh#L9

Reply:
For CRI-O we do not need libsystemd.a, which made things a bit easier. Generally we can refer any package path in the nix store via `${package-name}`, so for example `${glibc.static}` will output `/nix/store/f1bk23abhim7fsqwv2g1n2z7gsga8rd5-glibc-2.30-static`:

```
> ls /nix/store/f1bk23abhim7fsqwv2g1n2z7gsga8rd5-glibc-2.30-static/lib
libanl.a           libc.a      libdl.a  libm-2.30.a  libmcheck.a  libnsl.a      libresolv.a  librt.a
libBrokenLocale.a  libcrypt.a  libg.a   libm.a       libmvec.a    libpthread.a  librpcsvc.a  libutil.a
```